### PR TITLE
pyMipmap improvements

### PR DESCRIPTION
**Pull request by Hoikas**
This changeset adds more exception translators to the `pyMipmap` member functions and removes some redundant buffer copying.

### CODE REVIEW

File: Python/PRP/Surface/pyMipmap.cpp [hunk 1]
```diff
@@ -142,7 +142,12 @@ PY_METHOD_VA(Mipmap, setLevel,
         PyErr_SetString(PyExc_TypeError, "setLevel expects int, binary string");
         return NULL;
     }
-    self->fThis->setLevelData(level, (const void*)data, dataSize);
+    try {
+        self->fThis->setLevelData(level, (const void*)data, dataSize);
+    } catch (hsBadParamException& ex) {
```
Comment (zrax) [inline Python/PRP/Surface/pyMipmap.cpp]: Should catch by `const` reference

File: Python/PRP/Surface/pyMipmap.cpp [hunk 2]
```diff
@@ -216,21 +231,19 @@ PY_METHOD_NOARGS(Mipmap, extractColorData,
     "Extract an RGB color buffer from a JPEG mipmap")
 {
     size_t dataSize = self->fThis->getWidth() * self->fThis->getHeight() * 3;
-    char* data = new char[dataSize];
+    PyObject* buf = PyBytes_FromStringAndSize(NULL, dataSize);
+    char* data = PyBytes_AS_STRING(buf);
```
Comment (zrax) [inline Python/PRP/Surface/pyMipmap.cpp]: This either needs to be added to PyPlasma.h for Python 2.x, or we need to formally drop support for Python 2.6 (I'd prefer the latter, but it's a slightly larger and more unrelated change)

File: Python/PRP/Surface/pyMipmap.cpp (hunk 1 above)
Comment (Hoikas) [reply Python/PRP/Surface/pyMipmap.cpp]: Heh, I had it this way... Then noticed the rest of the file was catching by non-`const` reference. Fixed.

File: Python/PRP/Surface/pyMipmap.cpp (hunk 2 above)
Comment (Hoikas) [reply Python/PRP/Surface/pyMipmap.cpp]: Fixed
